walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,95 @@
+        Collections.sort(eventsByEndTime, Event.ORDER_BY_END_DESCENDING);
+        Collections.sort(eventsByStartTime, Event.ORDER_BY_START_ASCENDING);
+
+        Collection<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+        int prevEndTime = TimeRange.END_OF_DAY + 1;
+        
+        for (Event event : eventsByEndTime) {
+            /* Find attendees required to be at meeting also attending event. */
+            Collection<String> conflictAttendees = new HashSet<>(attendees);
+            Collection<String> eventAttendees = event.getAttendees();
+            conflictAttendees.retainAll(event.getAttendees());
+
+            /**
+             * For events that have conflicting attendees, add the time range 
+             * spanning from the last recorded end time to the current event's end 
+             * time to our output set if the duration fits within it. Move last end 
+             * time to current event's start time for the next event. 
+             */
+            if (conflictAttendees.size() > 0) {
+                TimeRange freeTime = TimeRange.fromStartEnd(event.getWhen().end(), prevEndTime, false);
+                if (duration <= freeTime.duration()) {
+                    meetingTimes.add(freeTime);
+                } 
+                prevEndTime = (event.getWhen().start() < prevEndTime) ? event.getWhen().start() : prevEndTime;
+            }
+        }
+
+        /* Check the time range from the start of the day to the first event's start time. */
+        if (prevEndTime > 0) {
+            TimeRange firstFreeTime = TimeRange.fromStartEnd(0, prevEndTime, false);
+            meetingTimes.add(firstFreeTime);
+        }
+
+        List<TimeRange> meetingTimesList = new ArrayList<TimeRange>(meetingTimes);
+        Collections.reverse(meetingTimesList);
+        return meetingTimesList;
+    }
+
+    /**
+     * The query function returns a set of TimeRanges where a potential meeting,
+     * specified as a MeetingRequest argument, can take place among a full day of 
+     * Events. The algorithm runs in O(n^2) time since we have to sort the events
+     * first, requiring O(nlogn) time, and then iterate through each event and 
+     * find conflicting attendees, requiring O(n^2) time.
+     */
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+        Collection<String> mandatoryAttendees = request.getAttendees();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
+        Collection<String> allAttendees = new ArrayList<>(mandatoryAttendees);
+        allAttendees.addAll(optionalAttendees);
+        
+        /* Check validity of arguments. */
+        long duration = request.getDuration();
+        if (duration < 0 || duration >= TimeRange.END_OF_DAY) {
+            return meetingTimes; // return empty if no duration or if duration is too long
```

Comment:
```suggestion
            return meetingTimes; // return empty if invalid duration or if duration is too long
```
or at least something saying that the duration is invalid instead of no duration